openshift-client/src/main/java/io/fabric8/openshift/client/DefaultOpenShiftClient.java

```diff
@@ -147,17 +146,22 @@ public DefaultOpenShiftClient(String masterUrl) throws KubernetesClientException
   protected DefaultOpenShiftClient(OkHttpClient httpClient, OpenShiftConfig config) throws KubernetesClientException {
     super(httpClient, config);
     try {
-      addOpenShiftAuthInterceptor();
-      this.delegate = new DefaultKubernetesClient(httpClient, config);
+      this.delegate = new DefaultKubernetesClient(clientWithOpenShiftOAuhtInterceptor(httpClient), config);
       this.openShiftUrl = new URL(config.getOpenShiftUrl());
     } catch (MalformedURLException e) {
       throw new KubernetesClientException("Could not create client", e);
     }
   }
 
-  private void addOpenShiftAuthInterceptor() {
-    httpClient = httpClient.newBuilder().authenticator(Authenticator.NONE).build();
-    httpClient = httpClient.newBuilder().addInterceptor(new OpenShiftOAuthInterceptor(httpClient,  OpenShiftConfig.wrap(getConfiguration()))).build();
```

Comment:
If we use the builder from the existing client, we are getting stuck with an un-modifiable list of interceptors. So, instead we can just create a new builder and control what pieces we want to keep from the old one. 

Regarding `swapable behavior`, I like the idea, but it is really overkill for the particular case.

Regarding `using a single interceptor`, why spoil something that is so cleanly separated?